File: src/main/java/com/ReachU/ServiceBookingSystem/services/company/AdminService.java
package com.ReachU.ServiceBookingSystem.services.company;


import com.ReachU.ServiceBookingSystem.dto.AdDTO;
import com.ReachU.ServiceBookingSystem.dto.ReservationDTO;
import com.ReachU.ServiceBookingSystem.entity.User;
import org.springframework.security.core.userdetails.UserDetails;

import java.io.IOException;
import java.util.List;

public interface AdminService {

//    Long getUserIdFromUserDetails(UserDetails userDetails);

    List<User> getAllUsers();

    boolean postService(Long userId, AdDTO adDTO) throws IOException;

    List<AdDTO> getAllAds(Long userId);

    AdDTO getAdById(Long adId);

    boolean updateAd(Long adId, AdDTO adDTO) throws IOException;

    boolean deleteAd(Long adId);

    List<ReservationDTO> getAllAdBookings(Long adminId);

    boolean changeBookingStatus(Long bookingId, String status);

    Long getUserIdFromUserDetails(UserDetails userDetails);

    Long getAdminIdFromUserDetails(UserDetails userDetails);

//    boolean postService(UserDetails userDetails, AdDTO adDTO) throws IOException;
}

File: src/app/admin/services/admin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Observable} from "rxjs";
import { UserStorageService } from 'src/app/auth/services/user-stoarge.service';

const BASIC_URL = "http://localhost:8080/";

@Injectable({
  providedIn: 'root'
})
export class AdminService {


    constructor(private http: HttpClient) { }

        postService(adDTO:any): Observable<any>{
            const userId = UserStorageService.getUserId();
            console.log("User Id is:",userId );
            return this.http.post(BASIC_URL + `api/admin/ad/${userId}`, adDTO, {
            // headers : this.createAuthorizationHeader()
            })
        }

    getAllAdsByUserId(): Observable<any>{
        const userId = UserStorageService.getUserId();
        return this.http.get(BASIC_URL + `api/admin/ads`, {
            // headers : this.createAuthorizationHeader()
        })
    }

    getAdById(adId:any): Observable<any>{
        return this.http.get(BASIC_URL + `api/admin/ad/${adId}`, {
            // headers : this.createAuthorizationHeader()
        })
    }

    updateAd(adId:any, adDTO:any): Observable<any>{
        return this.http.put(BASIC_URL + `api/admin/ad/${adId}`, adDTO, {
            // headers : this.createAuthorizationHeader()
        })
    }

    deletedAd(adId:any): Observable<any>{
        return this.http.delete(BASIC_URL + `api/admin/ad/${adId}`, {
            // headers : this.createAuthorizationHeader()
        })
    }

    getAllAdBookings(): Observable<any>{
        const adminId = UserStorageService.getUserId();
        return this.http.get(BASIC_URL + `api/admin/bookings/${adminId}`, {
            // headers : this.createAuthorizationHeader()
        })
    }

    changeBookingStatus(bookingId: number, status: string): Observable<any>{

        return this.http.get(BASIC_URL + `api/admin/booking/${bookingId}/${status}`, {
            // headers : this.createAuthorizationHeader()
        })
    }

    getAllCategories(): Observable<any> {
        return this.http.get(BASIC_URL + 'api/admin/categories',{
            // headers: this.createAuthorizationHeader()
        })
    }

    getTotalOrders(): Observable<any>  {
        return this.http.get<any>(BASIC_URL+ 'api/admin/dashboard/total-orders', {
        })
    }

    getTotalCustomers(): Observable<any> {
        return this.http.get<any>(BASIC_URL+ 'api/admin/dashboard/total-customers', {
        })
    }

    getTotalRevenue(): Observable<any>  {
       return this.http.get<any>(BASIC_URL+ 'api/admin/dashboard/total-revenue', {
        })
    }

    getTotalBookings(): Observable<any>  {
        return this.http.get<any>(BASIC_URL+ 'api/admin/dashboard/total-bookings', {
        })
    }
}

